### PR TITLE
(feat) O3-2618: Attachments not hooked up in Visit Note

### PR DESCRIPTION
**Pull request by brandones**
## Requirements

- [x] This PR has a title that briefly describes the work done including the ticket number. If there is a ticket, make sure your PR title includes a [conventional commit](https://o3-dev.docs.openmrs.org/#/getting_started/contributing?id=your-pr-title-should-indicate-the-type-of-change-it-is) label. See existing PR titles for inspiration.
- [ ] My work conforms to the [OpenMRS 3.0 Styleguide](https://om.rs/styleguide) and [design documentation](https://zeroheight.com/23a080e38/p/880723-introduction).
- [ ] My work includes tests or is validated by existing tests.

## Summary

This makes the "Add image" button in the visit note work. It launches the attachments modal. If an image is uploaded, it will be previewed below the button. The button changes to "Change image" if there is one already.

Potentially controversial: it makes esm-patient-notes-app peer-depend on esm-patient-attachments-app. I think this makes sense, though, given that the latter provides the modal.

There are no designs for this feature other than the button, so I improvised. See [designs](https://app.zeplin.io/project/60d59321e8100b0324762e05/screen/60d5e3ada2a468bcf05bdb8c).

## Screenshots

https://github.com/openmrs/openmrs-esm-patient-chart/assets/1031876/2bb8454f-b29b-4d4d-b6e9-2e45ce60a147

## Related Issue

https://openmrs.atlassian.net/browse/O3-2618

### CODE REVIEW

**Comment by vasharma05** [comment]
We still haven't figured out how the attachment added to the visit note is related to the visit note itself. If we go to the visit summary, we should have the reference to the attachment added.
CC: @ibacher @gracepotma @brandones

File: packages/esm-patient-notes-app/package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
   },
   "peerDependencies": {
     "@openmrs/esm-framework": "5.x",
+    "@openmrs/esm-patient-attachments-app": "5.x",
```
[the remark this answers is not in the document]
Comment (brandones) [reply packages/esm-patient-notes-app/package.json:45]: Yeah, good call @ibacher, much better idea. `esm-api` actually does seem like an appropriate place for this.

[the remark this answers is not in the document]
Comment (jayasanka-sack) [reply packages/esm-patient-notes-app/package.json:45]: Hello @brandones! Is this pull request ready for merging, or should we wait until the above migration is completed?

[the remark this answers is not in the document]
Comment (brandones) [reply packages/esm-patient-notes-app/package.json:45]: Sorry for the delay @jayasanka-sack . I'm going to do what Ian described here and update this PR.

[the remark this answers is not in the document]
Comment (brandones) [reply packages/esm-patient-notes-app/package.json:45]: Made the core change: https://github.com/openmrs/openmrs-esm-core/pull/860

**Comment by denniskigen** [comment]
@ibacher, could you have a look at this again?

**Comment by denniskigen** [comment]
Hey @brandones, I encountered a `413: Entity Too Large error` while uploading an attachment for the first time. The JPEG image I uploaded is 3.8 MB, which should be within the 10 MB size limit suggested by the attachment upload modal. Do you know whether the API imposes a lower size limit than that? If so, we should update the helper text to reflect the actual size limit imposed by the API.

Also, ignore the 502 error in the video - the RefApp randomly went down during my local testing.

https://github.com/openmrs/openmrs-esm-patient-chart/assets/8509731/a1496933-442a-4432-927b-e23ab4d9d308

**Comment by denniskigen** [comment]
Second time round, the upload worked like a charm when using a smaller image (122 KB). I could see the attachment and encounter posted but the encounter doesn't show up in the Visits dashboard. Perhaps its something completely unrelated to the upload functionality? I'm investigating..

https://github.com/openmrs/openmrs-esm-patient-chart/assets/8509731/013d6548-5eed-4cee-85ed-d1f222a2efd3

The attachment loads up fine in the Attachments dashboard:

CleanShot 2024-01-24 at 12  44 49@2x

**Review by ibacher — APPROVED** [review]

**Comment by ibacher** [comment]
IIRC, Tomcat's default configuration restricts the upload size to 2 MB. That's modifiable, but there's little reason to do that on dev3. There are actually two relevant GPs for this `attachments.maxUploadFileSize` and `attachments.maxStorageFileSize`. I don't really know why we have those or why the second is (by default) so much smaller than the first, but it would probably be good (new ticket) if the frontend just respected `attachments.maxUploadFileSize` rather than having it's own property.

**Comment by denniskigen** [comment]
Ah, thanks for the context, @ibacher. I'll also file one about updating the helper text in the modal that's currently indicating that the limit is 10 MB. I assume that'll default to 2 MB then?

**Review by denniskigen — APPROVED** [review]
Nice work, @brandones!

**Comment by denniskigen** [comment]
@ibacher, should you need to have an active visit open to record a visit note?

**Comment by ibacher** [comment]
> I'll also file one about updating the helper text in the modal that's currently indicating that the limit is 10 MB. I assume that'll default to 2 MB then?

Yeah, that's probably fair.

> should you need to have an active visit open to record a visit note?

Absolutely! What is a visit note if it's not attached to some kind of visit?

**Comment by denniskigen** [comment]
Well, currently you can record a visit note without a visit. But fixing that is trivial.

**Comment by brandones** [comment]
Thanks guys!

**Comment by denniskigen** [comment]
FYI: The attachments modal isn't loading up correctly on dev3 at the moment? Could it be a bad deploy from Bamboo, @ibacher? I've been seeing some errors about webpack chunks in the last couple days.

**Comment by denniskigen** [comment]
Working fine now, after triggering a redeployment. Dunno if that was the root cause.

**Pull request merged**